src/model.jl

```diff
@@ -499,89 +549,120 @@ function set_capacity_cost(m, n, 𝒯, t_inv,  modeltype::EnergyModel, ::Rolling
+function set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype, ::UnlimitedLife)
+    # Deduce the required variables
+    var_capex = get_var_capex(m, prefix, n)
+    var_rem = get_var_rem(m, prefix, n)
+
     # The capacity has an unlimited lifetime, one investment at the beginning of t_inv
-    capex_val = capex(n, t_inv)
-    @constraint(m, m[:capex_stor][n, t_inv] == capex_val.level * m[:stor_cap_add][n, t_inv])
-    @constraint(m, m[:capex_rate][n, t_inv] == capex_val.rate * m[:stor_rate_add][n, t_inv])
-    @constraint(m, m[:stor_cap_rem][n, t_inv] == 0)
-    @constraint(m, m[:stor_rate_rem][n, t_inv] == 0)
+    capex_val = set_capex_value(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ)
+    @constraint(m, [t_inv ∈ 𝒯ᴵⁿᵛ], var_capex[t_inv] == capex_val[t_inv])
+
+    # Fix the binary variable
+    for t_inv ∈ 𝒯ᴵⁿᵛ
+        fix(var_rem[t_inv], 0; force=true)
+    end
 end
 
-function set_capacity_cost(m, n::Storage, 𝒯, t_inv,  modeltype::EnergyModel, ::StudyLife)
+function set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype, ::StudyLife)
+    # Deduce the required variables
+    var_capex = get_var_capex(m, prefix, n)
+    var_rem = get_var_rem(m, prefix, n)
+
     # The capacity is limited to the end of the study. Reinvestments are included
     # No capacity removed as there are reinvestments according to the study length
-    capex_val = capex(n, t_inv)
-    stor_capex = capex_val.level * set_capex_value(remaining(t_inv, 𝒯), lifetime(n, t_inv), discount_rate(modeltype))
-    rate_capex = capex_val.rate * set_capex_value(remaining(t_inv, 𝒯), lifetime(n, t_inv), discount_rate(modeltype))
-    @constraint(m, m[:capex_stor][n, t_inv] == stor_capex * m[:stor_cap_add][n, t_inv])
-    @constraint(m, m[:capex_rate][n, t_inv] == rate_capex * m[:stor_rate_add][n, t_inv])
-    @constraint(m, m[:stor_cap_rem][n, t_inv] == 0)
-    @constraint(m, m[:stor_rate_rem][n, t_inv] == 0)
+    capex_disc = StrategicProfile(
+        [
+            set_capex_discounter(
+                remaining(t_inv, 𝒯ᴵⁿᵛ),
+                lifetime(inv_data, t_inv),
+                discount_rate(modeltype)
+            ) for t_inv ∈ 𝒯ᴵⁿᵛ
+        ]
+    )
+    capex_val = set_capex_value(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ)
+    @constraint(m, [t_inv ∈ 𝒯ᴵⁿᵛ], var_capex[t_inv] == capex_val[t_inv] * capex_disc[t_inv])
+
+    # Fix the binary variable
+    for t_inv ∈ 𝒯ᴵⁿᵛ
+        fix(var_rem[t_inv], 0; force=true)
+    end
 end
 
-function set_capacity_cost(m, n::Storage, 𝒯, t_inv,  modeltype::EnergyModel, ::PeriodLife)
+function set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ,  modeltype, ::PeriodLife)
+    # Deduce the required variables
+    var_capex = get_var_capex(m, prefix, n)
+    var_add = get_var_add(m, prefix, n)
+    var_rem = get_var_rem(m, prefix, n)
+
     # The capacity is limited to the current sp. It has to be removed in the next sp.
     # The formula for capacity updating uses the cap_rem for the previous sp, hence the sps used here.
```

Comment:
I think I added it to the documentation in the last PR, but I am uncertain. I will double check.

Reply:
I will earmark this as well for the last commit.

Reply:
We have in the documentation a description saying that capacity additions are occurring at the beginning of a strategic period, that is the capacity is available in the strategic period in which the investment is happening, and removing is occurring at the end of a strategic period. In that respect, I think it would be too much to add it here as well. However, I rephrased it in the last PR.